### PR TITLE
Add table truncate feature

### PR DESCRIPTION
**Pull request by cvogt**
review by @szeiger

This needs to be reopened against master after action-monad was merged. Just opening this now to allow review.

### CODE REVIEW

File: src/main/scala/scala/slick/driver/JdbcActionComponent.scala [hunk 1]
```diff
@@ -174,20 +174,26 @@ trait JdbcActionComponent extends SqlActionComponent { driver: JdbcDriver =>
   def createSchemaActionExtensionMethods(schema: SchemaDescription): SchemaActionExtensionMethods =
     new SchemaActionExtensionMethodsImpl(schema)
 
-  class SchemaActionExtensionMethodsImpl(schema: SchemaDescription) extends super.SchemaActionExtensionMethodsImpl {
-    def create: DriverAction[Effect.Schema, Unit] = new DriverAction[Effect.Schema, Unit] with JdbcDatabaseAction[Unit] {
-      def statements = schema.createStatements
+  private def mkAction[E <: Effect](dumpInfoKey: String, _statements: Iterator[String])
```
Comment (szeiger) [inline src/main/scala/scala/slick/driver/JdbcActionComponent.scala]: should probably be rolled into SimpleJdbcAction (which is not a SqlAction at the moment)

File: src/main/scala/scala/slick/driver/JdbcActionComponent.scala [hunk 2]
```diff
@@ -174,20 +174,26 @@ trait JdbcActionComponent extends SqlActionComponent { driver: JdbcDriver =>
...
 
-    def drop: DriverAction[Effect.Schema, Unit] = new DriverAction[Effect.Schema, Unit] with JdbcDatabaseAction[Unit] {
-      def statements = schema.dropStatements
-      def run(ctx: ActionContext[Backend]): Unit =
-        for(s <- statements) ctx.session.withPreparedStatement(s)(_.execute)
-      override def getDumpInfo = super.getDumpInfo.copy(name = "schema.drop")
-    }
+  final def truncateTableAction(table: Table[_]): DriverAction[Effect.Write,Unit]
+    = mkAction[Effect.Write](
+        "write.truncate",
```
Comment (szeiger) [inline src/main/scala/scala/slick/driver/JdbcActionComponent.scala]: "schema.truncate"?

File: src/main/scala/scala/slick/profile/RelationalProfile.scala [hunk 3]
```diff
@@ -70,6 +70,8 @@ trait RelationalProfile extends BasicProfile with RelationalTableComponent
     //@deprecated("Use .schema instead of .ddl", "2.2")
     final def ddl: SchemaDescription = schema
 
+    final def truncate: DriverAction[Effect.Write, Unit] = truncateTableAction(q.shaped.value)
```
Comment (szeiger) [inline src/main/scala/scala/slick/profile/RelationalProfile.scala]: Needs a doc comment

**Comment by scspijker** [comment]
:+1:

**Comment by direction** [comment]
:+1:

**Comment by cvogt** [comment]
@trevorsibanda maybe you can port this to current master?

**Comment by trevorsibanda** [comment]
@cvogt will give it a shot

**Comment by cvogt** [comment]
fixed in #1591

**Pull request closed without merging**